firefox-ios/Storage/Rust/RustLogins.swift

```
@@ -829,6 +830,30 @@ public class RustLogins: LoginsProtocol {
         return deferred
     }
 
+    func updateLogin(id: String, login: LoginEntry, completionHandler: @escaping () -> Void) {
+        queue.async {
+            guard self.isOpen else {
+                _ = LoginsStoreError.UnexpectedLoginsApiError(reason: "Database is closed")
+                completionHandler()
+                return
+            }
+
+            self.getStoredKey { result in
+                switch result {
+                case .success(let key):
+                    do {
+                        _ = try self.storage?.update(id: id, login: login, encryptionKey: key)
+                        completionHandler()
+                    } catch _ as NSError {
+                        completionHandler()
+                    }
+                case .failure:
+                    completionHandler()
```

Same for those two completion handlers, we can return an error